account_payment_partner/models/account_move.py

```diff
@@ -65,6 +66,18 @@ def _compute_payment_mode(self):
                     move.payment_mode_id = move.with_context(
                         force_company=move.company_id.id
                     ).partner_id.customer_payment_mode_id
+                elif (
+                    move.type in ["out_refund", "in_refund"] and move.reversed_entry_id
+                ):
+                    move.payment_mode_id = move.reversed_entry_id.payment_mode_id
```

Comment:
Is this executed when doing the reverse?

Reply:
Now, some tests check it.